Add Course theme

donnapep opened the pull request:
#### Changes proposed in this Pull Request:
This theme is meant to be used with [Sensei LMS](https://senseilms.com/), but can also be used standalone. We are moving the code from the https://github.com/Automattic/course repo to here so that theme updates can be deployed to both WordPress.org and WordPress.com.

Replaces https://github.com/Automattic/themes/pull/6766. The original feedback left there should be resolved in this PR.

donnapep committed:
Add Course theme

This theme is meant to be used with Sensei LMS, but can also be used standalone.

jffng commented on course/course.pot:
```diff
@@ -0,0 +1,390 @@
+# Copyright (C) 2023 Automattic
```
Typically we do not commit the .pot files here, they just live on wpcom. Is there a reason to include them here?

donnapep replied:
I've removed it in https://github.com/Automattic/themes/pull/6888/commits/23dde8b383be94b40683307187bf5ec567a4908f.

jffng commented on course/style.css:
```diff
@@ -0,0 +1,429 @@
+ * spacing defined above. This behavior may be built into
+ * the Block Editor in the future.
+ */
+.wp-block-navigation__responsive-container.is-menu-open {
```
This CSS can be removed now I think.

donnapep replied:
Yup! Tested and removed in https://github.com/Automattic/themes/pull/6888/commits/1ceea814319056ae138f94ba97d0a0750808e534.

jffng commented on course/style.css:
```diff
@@ -0,0 +1,429 @@
+/*
+ * Search styles
+ */
+.wp-block-search__button {
```
What's the reason for setting this in CSS? 

This is what the default search button looks like when I insert a search block:
 
<img width="658" alt="Screenshot 2023-03-03 at 11 27 14 AM" src="https://user-images.githubusercontent.com/5375500/222774269-76ed68aa-8045-4236-a3b1-53a8025b9d87.png">

donnapep replied:
I wasn't able to reproduce that issue, but when I removed the CSS it didn't appear to break anything either - [1bacf4a](https://github.com/Automattic/themes/pull/6888/commits/1bacf4a63262fc9e7f5d3d72a551886b5609284a).

jffng commented on course/patterns/comments.php:
```diff
@@ -0,0 +1,49 @@
+<!-- wp:comments-pagination-next /-->
+<!-- /wp:comments-pagination -->
+
+<!-- wp:post-comments-form /--></div>
```
Noticing an issue with the submit button for the comment form: 

<img width="1070" alt="Screenshot 2023-03-03 at 11 34 23 AM" src="https://user-images.githubusercontent.com/5375500/222775850-ec0a3ca7-d1a5-4e5b-bf06-4f698e682105.png">

donnapep replied:
@jffng Hmmm...I can't reproduce this one either when I tried on a Jurassic Ninja site:

![Screenshot 2023-03-06 at 8 38 25 AM](https://user-images.githubusercontent.com/1190420/223125826-71530892-109d-42c4-95e7-52eae65ff38e.jpg)

jffng reviewed (commented):
This is a nice theme! I left a few comments of some issues I noticed, but generally looking good.

donnapep committed:
Remove .pot file

donnapep committed:
Remove obsolete CSS

donnapep committed:
Remove color on search button

donnapep commented:
@jffng Any update on this one? We have some more changes ready for this theme that are blocked by this PR.

jffng commented on course/patterns/footer.php:
```diff
@@ -0,0 +1,173 @@
+      <p class="has-text-align-center has-system-font-family has-x-small-font-size" style="margin-top:var(--wp--preset--spacing--70);padding-top:var(--wp--preset--spacing--50);padding-bottom:5rem;letter-spacing:0.02em">
+        <?php
+        // Supports Footer Credit cusotmization on WPCOM and WoA sites.
+        $is_atomic = ( defined( 'ATOMIC_SITE_ID' ) && ATOMIC_SITE_ID );
```
We try to avoid wpcom specific customization like this in themes, is there a way to remove this? I believe on simple sites we don't do anything like this and they are still able to show a custom footer.

donnapep replied:
> We try to avoid wpcom specific customization like this in themes, is there a way to remove this?

@mikeyarce Would you mind chiming in on this as I believe you added the footer credit?

Also, while I have you here, any idea about the answers to these other questions? 🙂 

> Will the theme be available on the wpcom themes showcase as a free theme also?

> It looks like the theme is already in pub, to confirm there is no code there that will conflict with this version?

mikeyarce replied:
The footer credit was customized because of how the WordPress.com Footer works - there is a specific regex in place that will not replace the custom footer we have on Sensei, which is why I opted to have a customization there.

If we want to remove our custom Sensei footer, then we don't need the specific wpcom customization.

> Will the theme be available on the wpcom themes showcase as a free theme also?

I'm not sure - I think it's up to the themes team what they want to do with it?  I would say it's probably best to make it a free theme.

>  It looks like the theme is already in pub, to confirm there is no code there that will conflict with this version?

The theme we commited to `pub` is a bit older, the only changes we would need to keep from the already committed one are the Headstart Annotations.

jffng replied:
> If we want to remove our custom Sensei footer, then we don't need the specific wpcom customization.

👍  That would be best to remove, then I think this is good to go.

Thanks!

donnapep replied:
@mikeyarce Is this something you'd prefer to handle? Feel free to commit directly to this PR.

mikeyarce replied:
Done in https://github.com/Automattic/themes/pull/6888/commits/e73455c1621677cb4b2a6e3fab3acde606c46984 😄

jffng reviewed (commented):
Hi @donnapep apologies for the delay. I think I'm missing some context. Will the theme be available on the wpcom themes showcase as a free theme also? 

> We are moving the code from the https://github.com/Automattic/course repo to here so that theme updates can be deployed to both WordPress.org and WordPress.com.

It looks like the theme is already in `pub`, to confirm there is no code there that will conflict with this version? 

Thanks for bearing with me, don't mean to gatekeep, just want to make sure we're in the right spot :)

jffng reviewed (commented):
There are a few more pieces that should be changed before this can be launched on dotcom.

The home page template does not match the screenshot. We have been using `home.html` to achieve this consistency. It's worth noting that introducing this template will potentially change / break the home page for existing users of this theme.

![image](https://user-images.githubusercontent.com/5375500/225681938-723ec720-a28f-4971-9d69-713224890522.png)

Line-height is off in Large and Extra-large sizes.

![image](https://user-images.githubusercontent.com/5375500/225682758-b5e55f5e-9cc1-4ff8-8572-6fb556ee05c0.png)

The gallery block captions line-height should be corrected, and the type treatment makes it difficult to read.

![image](https://user-images.githubusercontent.com/5375500/225682523-1012b3da-8c90-4734-892d-90da217f08fc.png)

mikeyarce committed:
Remove custom Sensei footer.

jffng commented on course/theme.json:
```diff
@@ -0,0 +1,923 @@
+		"typography": {
+			"fontFamily": "var(--wp--preset--font-family--body)",
+			"fontSize": "var(--wp--custom--typography--font-sizes--normal)",
+			"lineHeight": "1.838rem"
```
Could we use a unitless value here instead? Using rem here doesn't work well for the larger font sizes.

donnapep replied:
I removed this in https://github.com/Automattic/themes/pull/6888/commits/2217dfebb24a0e9b8d1fdd089f66819cc78e7245.

jffng commented on course/theme.json:
```diff
@@ -0,0 +1,923 @@
+					"blockGap": "20px"
+				},
+				"typography": {
+					"lineHeight": "clamp(1.838rem, 1.763rem + 0.249vw, 1.95rem)",
```
Could we use a unitless value here instead? Using rem here doesn't work well for the larger font sizes.

donnapep replied:
This is gone too in https://github.com/Automattic/themes/pull/6888/commits/2217dfebb24a0e9b8d1fdd089f66819cc78e7245 as it's not needed.

jffng commented on course/style.css:
```diff
@@ -0,0 +1,412 @@
+	text-align: center;
+}
+
+ /* Background and text color, padding around for navbar, only for mobile view. Desktop view has separate color */
```
There is a lot of navigation specific CSS, do you think this kind of customization should be available via the UI? If so, could we suggest issues in Gutenberg to improve or fix the issues?

donnapep replied:
Yes, perhaps. I didn't work on the navigation bit so it's unclear to me if some or all of this could be moved into `theme.json`. My preference would be to do any refactoring outside of this particular PR though.

jom committed:
Fix path of the Inter font

On the default and `blue` style.

donnapep committed:
Fix line height

donnapep committed:
Make image captions smaller than post content

donnapep commented:
> The home page template does not match the screenshot.

I don't think we're too keen on changing the screenshot to match the home page. While it's true that the Course theme can be used without Sensei, we want to do everything we can to promote its use with Sensei. We consider it the primary use case.

donnapep commented:
@jffng Thanks for the review! I believe I've addressed your feedback or else responded with comments to some of your other feedback. Please let me know what you think. 🙂

m1r0 commented:
This has become a blocker for us so I'm approving.

Any additional feedback from the Themes team is very welcome and we can address that in a separate PR. 🙏

jffng commented:
@donnapep @m1r0 apologies for the delay, yes please go ahead and merge if it meets your requirements. Docs for deploying the theme to wpcom can be found here: PNEWy-eqL-p2

As a follow up, we can also add the theme to the list of themes to automated deploy for the dotorg repo here: https://github.com/Automattic/themes/blob/trunk/deploy-dotorg.sh#L18

I also want to clarify in the future, please don't wait on @Automattic/theam to approve or merge your PRs, especially if it's become a blocker. We're happy to provide feedback and perspective. We're stewards of this repo but cannot be DRIs or approve every PR for all the themes here, since in many cases we do not have all the context and specific requirements.
